org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/SbtScopesBuildManager.scala

```diff
@@ -52,34 +58,45 @@ class SbtScopesBuildManager(val owningProject: IScalaProject, managerSettings: S
     }
   }
 
+  override def buildErrors: Set[IMarker] = buildScopeUnits.flatMap { _.buildErrors }.toSet
   override def invalidateAfterLoad: Boolean = true
   override def clean(implicit monitor: IProgressMonitor): Unit = buildScopeUnits.foreach { _.clean }
   override def canTrackDependencies: Boolean = true
 
   private def findProjectsInError(scopeUnit: BuildScopeUnit) = {
-    def hasErrors(project: IProject, scope: CompileScope): Boolean = {
-      IScalaPlugin().getScalaProject(project).buildManager match {
-        case manager: SbtScopesBuildManager => manager.hasErrors(scope)
-        case manager: EclipseBuildManager => manager.hasErrors
-      }
-    }
+    def hasErrors(project: IProject, scope: CompileScope): Boolean =
+      IScalaPlugin().asScalaProject(project).map {
+        _.buildManager match {
+          case manager: SbtScopesBuildManager => manager.hasErrors(scope)
+          case manager: EclipseBuildManager => manager.hasErrors
+        }
+      }.getOrElse(false)
     for {
       scope <- scopeUnit.scope.dependentScopesInUpstreamProjects
       project <- owningProject.transitiveDependencies if hasErrors(project, scope)
     } yield project
   }
 
-  private def isStopBuildOnErrors = {
+  private def doesContinueBuildOnErrors = {
     val stopBuildOnErrorsProperty = SettingConverterUtil.convertNameToProperty(ScalaPluginSettings.stopBuildOnErrors.name)
-    owningProject.storage.getBoolean(stopBuildOnErrorsProperty)
+    !owningProject.storage.getBoolean(stopBuildOnErrorsProperty)
   }
 
   private def putMarkersForTransitives(scopeWithError: ScopeUnitWithProjectsInError): Unit = {
     if (scopeWithError.projectsInError.nonEmpty) {
       val errorProjects = scopeWithError.projectsInError.map(_.getName).toSet.mkString(", ")
+      val rootErrors = scopeWithError.projectsInError.flatMap { project =>
+        IScalaPlugin().asScalaProject(project).toList.flatMap {
+          _.buildManager.buildErrors
+        }
+      }.toSet[IMarker].map {
```

Comment:
Generally, I would prefer to have more vals here instead of this method chaining. Makes debugging easier as long as our debugger can't jump easily between lambdas. But just a nitpick.